src/librbd/io/AioCompletion.cc

```diff
@@ -161,12 +165,17 @@ void AioCompletion::complete_request(ssize_t r)
   assert(ictx != nullptr);
   CephContext *cct = ictx->cct;
 
+  if (r < 0 && r <= -MAX_ERRNO) {
```

Comment:
Nit: unneeded once you add an optional pointer in the API for returning the mismatch offset